### PR TITLE
Create gem-push.yml workflow

### PR DESCRIPTION
**Pull request by samsarkleio**
Initial workflow for pushing to rubygems.

This creates a workflow that is triggered after a release is published on github. It uses a token created specifically for github actions to be able to push a gem to rubygems.org.

### CODE REVIEW

File: .github/workflows/gem-push.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+name: Ruby Gem
+
+on:
+#   push:
+#     branches: [ master ]
+  # run manually for now
+  workflow_dispatch:
```
Comment (alecclarke) [inline .github/workflows/gem-push.yml:7]: Eventually, could we have this linked to creating on publishing a new release?
https://docs.github.com/en/actions/reference/events-that-trigger-workflows#release

Out of curiosity, how can we run actions manually? I've only had experience with automatically triggered ones.

File: .github/workflows/gem-push.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    - name: Publish to RubyGems
+      run: |
+        mkdir -p $HOME/.gem
+        touch $HOME/.gem/credentials
+        chmod 0600 $HOME/.gem/credentials
+        printf -- "---\n:rubygems_api_key: ${GEM_HOST_API_KEY}\n" > $HOME/.gem/credentials
+        gem build *.gemspec
+        gem push *.gem
+      env:
+        GEM_HOST_API_KEY: "${{secrets.RUBYGEMS_AUTH_TOKEN}}"
```
Comment (alecclarke) [inline .github/workflows/gem-push.yml:32]: Nitpicking, but could we add the `env` line above the `run`? I find it slightly easier to read the `GEM_HOST_API_KEY ` being set, then how it's used.

File: .github/workflows/gem-push.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+#     branches: [ master ]
+  # run manually for now
+  workflow_dispatch:
+
+jobs:
+  build:
+    name: Build + Publish
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
```
Comment (alecclarke) [inline .github/workflows/gem-push.yml:14]: Not a required change, but I believe @npaufler set the default Actions permission to `read`, so I think we get `read` by default.

File: .github/workflows/gem-push.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+      uses: actions/setup-ruby@v1
+      with:
+        ruby-version: 2.6.x
+
+    - name: Publish to RubyGems
+      run: |
+        mkdir -p $HOME/.gem
+        touch $HOME/.gem/credentials
+        chmod 0600 $HOME/.gem/credentials
+        printf -- "---\n:rubygems_api_key: ${GEM_HOST_API_KEY}\n" > $HOME/.gem/credentials
```
Comment (alecclarke) [inline .github/workflows/gem-push.yml:28]: Will this log the token to the Action's artifacts? If so, can it be used maliciously?

Comment (peerkleio) [reply .github/workflows/gem-push.yml:28]: Its a push only API token and you can't replace an existing gem version in rubygems.org.  This should not end up in artifacts from what I understand.  This is literally the supplied code from GitHub for implementing this action.

Comment (samsarkleio) [reply .github/workflows/gem-push.yml:28]: The above line puts the credentials in a file, it won't be logged/printed anywhere.

File: .github/workflows/gem-push.yml (hunk 2 above)
Comment (samsarkleio) [reply .github/workflows/gem-push.yml]: Fixed

File: .github/workflows/gem-push.yml (hunk 1 above)
Comment (samsarkleio) [reply .github/workflows/gem-push.yml]: Changed to be run after a published release. 

Before, to run it manually, we'd have to go to the "Actions" tab, select the workflow, and click "Run Manually". I wanted to have this for testing, but I think it's straightforward enough to just try a release.

File: .github/workflows/gem-push.yml (hunk 4 above)
Comment (alecclarke) [reply .github/workflows/gem-push.yml:28]: Ah! Thanks for the explanation! That was the piece I was missing.

**Review by alecclarke — APPROVED** [review]
Thank you for adding this in! It will make working on our public gems much easier ❤️

**Review by peerkleio — APPROVED** [review]

**Pull request merged**